src/core/Elsa.Core/Handlers/PersistWorkflow.cs

```diff
@@ -27,7 +27,7 @@ public async Task Handle(WorkflowExecutionPassCompleted notification, Cancellati
         {
             var behavior = notification.WorkflowExecutionContext.WorkflowBlueprint.PersistenceBehavior;
 
-            if (behavior == WorkflowPersistenceBehavior.ActivityExecuted || notification.ActivityExecutionContext.ActivityBlueprint.PersistWorkflow)
+            if (behavior == WorkflowPersistenceBehavior.ActivityExecuted || behavior == WorkflowPersistenceBehavior.WorkflowPassCompleted || notification.ActivityExecutionContext.ActivityBlueprint.PersistWorkflow)
```

Comment:
Aha. So, I think this may have been the reason I removed `WorkflowPassCompleted` in the past - it has effectively the same result as `ActivityExecuted`.

That being said, let's keep your changes in case we want to actually make a difference between the two behaviors.